poshc2/server/Payloads.py

```diff
@@ -223,7 +217,9 @@ def PatchBytes(self, filename, dll, offset, payloadtype, name=""):
             out = StringIO()
             with open("%spbind.ps1" % PayloadTemplatesDirectory, 'r') as f:
                 pbind = f.read()
-            pbind = str(pbind).replace("#REPLACEKEY#", self.Key)
+            pbind = str(pbind).replace("#REPLACEKEY#", self.Key) \
```

Comment:
Not necessarily for this commit, but at some point we should pull out all the `replace` calls into a single function e.g. `ReplacePayloadPlaceholders` or similar, which can be called for every payload (as e.g. replacing #REPLACEPBINDPIPENAME# will just not do anything if it doesn't exist in that particular template) then add unit tests for that function for various inputs which can actually be loaded from the `payload-templates` directory.

That test will then check that both the payload templates haven't been broken (e.g. some typos a placeholder) and the replace function doesn't break, as in either case it would fail.

Reply:
Yeah 100%, definitely a good way to go